### DISCUSSION TITLE
Inquiry Regarding Possible Alternatives for Male Long Deck Connector for Crazyflie

### DISCUSSION THREAD

**Discussion by wenhancao** [bitcraze/discussions#788] [General]
Dear Crazyflie Customer Support,

Recently, I've encountered a problem with my Crazyflie's Male Long Deck Connector as it has become bent. This has severely impacted the operation of my drone and I need to replace it. However, being a student, I am looking for a more affordable solution due to the substantial shipping fees and import taxes related to purchasing the connector directly from your store.

While looking for alternatives, I came across a product on Amazon with the following specifications:

Product Name: Straight Pin Headers
Pins: 40 Pins each Row
Row: One Row
Mounting Angle: Straight
Pin Pitch: 2.54mm
Pin Length: 25mm / 0.98"
Material: Plastic, Metal
Color: Black, Silver Tone
Weight: 119g
Package Content: 30 Pcs x Straight Pin Headers
I wanted to reach out to you to inquire if this particular product could serve as a compatible alternative for the Male Long Deck Connector. I am aware that this might not be an exact match, but I am hopeful that it might suit my needs.

In addition to that, I would also appreciate if you could provide any recommendations on the necessary specifications or even a direct purchase link for a suitable and more affordable connector. Your expert advice would be greatly appreciated.

**Comment by tobbeanton** [comment]
The pin headers with the Crazyflie is custom made and I think it will be hard to find a compatible replacement, mainly due to the length of them. If you do want to give it a try you should look for pin pitch of 2.0 mm, the 2.54mm will not fit.

Are you sure you can not bend the pins back?

You could check if any [local retailers](https://www.bitcraze.io/buy/local-retailers/) has the Male long deck connector.

**Reply by r-bt** [reply]
@tobbeanton I am also looking for the long male deck headers and am struggling to find a supplier given that there is no longer a USA based retailers. Do you know of any suppliers of these headers in the USA?

**Reply by ArisMorgens** [reply]
Hello @r-bt

Unfortunately, we are not aware of any suppliers of long pin headers in the USA. However, with a quick search you may be able to find some with specs close to the [original ones](https://www.bitcraze.io/documentation/hardware/male_longer_deck_connector/male_longer_deck_connector-datasheet.pdf).

**Reply by r-bt** [reply]
Hi @ArisMorgens Thanks for the quick reply. It looks like the specific combination of 2mm pitch and >18mm height makes these pins hard to find. For example on [Digikey](https://www.digikey.com/en/products/filter/headers-male-pins/314?s=N4IgjCBcoEwAwwtEBjKAzAhgGwM4FMAaEAeygG0QAWMAdgA4BWAThAF1iAHAFyhAGVuAJwCWAOwDmIAL7F4MGFFBpIWPEVIUQAZhi7tIYruZV6hnYzCJ2XXpAHDxU2SHlUlqDDgLEykSvSmrBwgPHyCopIyxAC0isgqwgCuGn6UjOzSWUA), there are no results for single row and double rows have limited or no stock.
